Index: youtrack-new-ui/app/pages/reports/_reports/types/toolbar.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- youtrack-new-ui/app/pages/reports/_reports/types/toolbar.html	(revision 5a3ee531b680fdc07d9c9d992af9ee2d925af199)
+++ youtrack-new-ui/app/pages/reports/_reports/types/toolbar.html	(revision )
@@ -5,7 +5,7 @@
                      report.lastCalculated.length > 0">
             <span translate>calculated</span><span> {{ report.lastCalculated }}</span>
         </span>
-        <span class="report-toolbar__calculated-date" ng-if="!report.lastCalculated || report.lastCalculated.length == 0">
+        <span class="report-toolbar__calculated-date" ng-if="isRecalculate()">
             <span  translate>calculating...</span>
         </span>
         <span ng-if="report.reportData && progress >= 0" class="report-toolbar__actions_progress">{{progress}}%</span>
Index: youtrack-new-ui/app/pages/reports/test/_reports/report.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- youtrack-new-ui/app/pages/reports/test/_reports/report.js	(revision 5a3ee531b680fdc07d9c9d992af9ee2d925af199)
+++ youtrack-new-ui/app/pages/reports/test/_reports/report.js	(revision )
@@ -338,6 +338,37 @@
         expect(loadReportHandler).toHaveBeenCalledWith(reportMock);
       });
 
+      it('should recalculate report', function() {
+        var recalculateHandler = jasmine.createSpy('onRecalculate');
+        $scope.id = 'testId';
+
+        expect($scope.isRecalculate()).toEqual(false);
+
+        $scope.recalculate(true).then(recalculateHandler);
+
+        expect($scope.isRecalculate()).toEqual(true);
+
+        reportRest.report.recalculate.defer.resolve({
+          name: 'ReportName',
+          state: 'CALCULATING'
+        });
+        $scope.$digest();
+
+        $timeout.flush();
+
+        reportMock.state = reportState.CALCULATING;
+        reportRest.report.reportWithData
+          .defer.resolve(reportMock);
+        $scope.$digest();
+
+        reportMock.state = reportState.READY;
+        $scope.$digest();
+        $timeout.flush();
+
+        expect(recalculateHandler).toHaveBeenCalled();
+        expect($scope.isRecalculate()).toEqual(false);
+      });
+
       it('should load report with old data if we pass force', function() {
         var recalculate = false;
         var force = true;
Index: youtrack-new-ui/app/pages/reports/_reports/report.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- youtrack-new-ui/app/pages/reports/_reports/report.js	(revision 5a3ee531b680fdc07d9c9d992af9ee2d925af199)
+++ youtrack-new-ui/app/pages/reports/_reports/report.js	(revision )
@@ -294,9 +294,22 @@
 
         scope.prepareReportRender = angular.noop;
 
+        scope._isRecalculate = false;
+
+        scope.isRecalculate = function() {
+          return scope._isRecalculate;
+        };
+
         scope.recalculate = function() {
           scope.setSpin(true);
-          return scope.loadReport(true);
+          scope._isRecalculate = true;
+
+          var onFinishRecalculation = function(response) {
+            scope._isRecalculate = false;
+            return response;
+          };
+
+          return scope.loadReport(true).finally(onFinishRecalculation);
         };
 
         scope.getDefaultRange = function() {
